internal/client/pagerduty_client.go

```diff
@@ -0,0 +1,167 @@
+//go:generate mockgen -package mocks --destination ../mocks/pagerduty_client.go . PagerdutyClient
+
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/base-org/pessimism/internal/logging"
+
+	"go.uber.org/zap"
+)
+
+// PagerdutyAction represents the type of actions that can be triggered by an event
+type PagerdutyAction string
+
+const (
+	Trigger                PagerdutyAction = "trigger"
+	PagerdutyAckAction     PagerdutyAction = "acknowledge"
+	PagerdutyResolveAction PagerdutyAction = "resolve"
+)
+
+// PagerdutySeverity represents the severity of an event
+type PagerdutySeverity string
+
+const (
+	Critical PagerdutySeverity = "critical"
+	Error    PagerdutySeverity = "error"
+	Warning  PagerdutySeverity = "warning"
+	Info     PagerdutySeverity = "info"
+)
+
+// PagerdutyResponseStatus is the response status of a Pagerduty API call
+type PagerdutyResponseStatus string
+
+const (
+	SuccessStatus PagerdutyResponseStatus = "success"
+)
+
+type PagerdutyConfig struct {
+	IntegrationKey  string
+	ChangeEventsURL string
+	AlertEventsURL  string
+}
+
+type PagerdutyClient interface {
+	PostEvent(ctx context.Context, event *PagerdutyEventTrigger) (*PagerdutyAPIResponse, error)
+}
+
+type pagerdutyClient struct {
+	integrationKey  string
+	changeEventsURL string
+	alertEventsURL  string
+	client          *http.Client
+}
+
+// NewPagerdutyClient ... Initializer for Pagerduty client
+func NewPagerdutyClient(cfg *PagerdutyConfig) PagerdutyClient {
+	if cfg.IntegrationKey == "" {
+		logging.NoContext().Warn("No Pagerduty integration key provided")
+	}
+
+	return pagerdutyClient{
+		integrationKey:  cfg.IntegrationKey,
+		changeEventsURL: cfg.ChangeEventsURL,
+		alertEventsURL:  cfg.AlertEventsURL,
+		client:          &http.Client{},
+	}
```

Comment:
Is there any reason why we're not returning a pointer?

Reply:
[6bb3dd7](https://github.com/base-org/pessimism/pull/130/commits/6bb3dd773305165e38a2d756dbf01f520c37075d)